core/app/models/spree/promotion_handler/free_shipping.rb

```diff
@@ -23,10 +23,12 @@ def activate
       private
 
       def promotions
-        Spree::Promotion.active.where(
-          id: Spree::Promotion::Actions::FreeShipping.pluck(:promotion_id),
-          path: nil
-        )
+        Spree::Promotion.active.
+          joins(:promotion_codes).where({
+            id: Spree::Promotion::Actions::FreeShipping.pluck(:promotion_id),
+            promotion_codes: { value: nil },
```

Comment:
I don't remember why I added this here, and it may just be the case that I'm too tired right now, but this certainly needs changing now that values cannot be `nil`/`NULL`